lib/fixed-points.nix

```diff
@@ -1,26 +1,66 @@
 { lib, ... }:
 rec {
   /*
-    Compute the fixed point of the given function `f`, which is usually an
-    attribute set that expects its final, non-recursive representation as an
-    argument:
+    `fix f` computes the fixed point of the given function `f`. In other words, the return value is `x` in `x = f x`.
```

Comment:
```suggestion
    `fix f` computes the fixed point of the given function `f`.
This is the value that satisfies `f x == x`.
```

"In other words" is the function definition, we already have that and it doesn't help.

What's missing here though is a non-magical explanation of how that value is found and when it works at all and why.

For instance, you can't say `fix (x: 1.0 / x)` although clearly the answer should be 1.

You also can't say `fix ({a,b,...}: {a=1;b=2;c=a+b;})` for reasons unclear to me.